src/services/productsService.js

```diff
@@ -23,6 +23,10 @@ class ProductsService {
 
     return product;
   }
+
+  async listAllProducts() {
+    return await this.productsRepo.getAll();
```

Comment:
notice that you `await` here and on the other side (in the router/controller). That means you are waiting for this promise to resolve before sending a new promise with the result of that promise to the caller. Also, if this promise rejects and throws an error, it will bubble up because you aren't catching here. 

If you were to write `return this.productsRepo.getAll()` instead, then you'd just pass along the original promise. If that promise rejects, it would only reject at the call site of this method.